tests/unit/generate_detect_target_contour.py

```diff
@@ -0,0 +1,226 @@
+"""
+Helper functions for `test_detect_target_contour.py`.
+
+"""
+
+import cv2
+import math
+import numpy as np
+
+from modules import detections_and_time
+
+
+LANDING_PAD_COLOUR_BLUE = (100, 50, 50)  # BGR
+
+
+class LandingPadImageConfig:
+    """
+    Represents the data required to define and generate a landing pad.
+    """
+
+    def __init__(
+        self,
+        centre: tuple[int, int],
+        axis: tuple[int, int],
+        blur: bool,
+        angle: float,
+    ):
+        """
+        centre: The pixel coordinates representing the centre of the landing pad.
+        axis: The pixel lengths of the semi-major axes of the ellipse.
+        blur: Indicates whether the landing pad should have a blur effect.
+        angle: The rotation angle of the landing pad in degrees clockwise, where 0.0 degrees
+            is where both semi major and minor are aligned with the x and y-axis respectively (0.0 <= angle <= 360.0).
+        """
+        self.centre = centre
+        self.axis = axis
+        self.blur = blur
+        self.angle = angle
+
+
+class NumpyImage:
+    """
+    Holds the numpy array which represents an image.
+    """
+
+    def __init__(self, image: np.ndarray):
+        """
+        image: A numpy array that represents the image.
+        """
+        self.image = image
+
+
+class BoundingBox:
+    """
+    Holds the data that define the generated bounding boxes.
+    """
+
+    def __init__(self, top_left: tuple[float, float], bottom_right: tuple[float, float]):
+        """
+        top_left: pixel coordinates representing the top left corner of the bounding box on an image.
+        bottom_right: pixel coordinates representing the bottom right corner of the bounding box on an image.
+        """
+        self.top_left = top_left
+        self.bottom_right = bottom_right
+
+
+class InputImageAndExpectedBoundingBoxes:
+    """
+    Struct to hold the data needed to perform the tests.
+    """
+
+    def __init__(self, image: np.ndarray, boxes_list: np.ndarray):
+        """
+        image: A numpy array that represents the image needed to be tested.
+        bounding_box_list: A numpy array that holds a list of expected bounding box coordinates.
+
+        The bounding box coordinates are in the following format:
+            top_left_x = boxes_list[0]
+            top_left_y = boxes_list[1]
+
+            bottom_right_x = boxes_list[2]
+            bottom_right_x = boxes_list[3]
```

Comment:
No, this is what a new member would write because they don't understand the existing documentation. Write the actual format.